code/modules/projectiles/guns/ballistic/automatic.dm

```diff
@@ -1380,6 +1382,43 @@
 	burst_shot_delay = 2.2
 	spread = 18
 
+/obj/item/gun/ballistic/automatic/assault_carbine/olympic
+	name = "OA-93 machine pistol"
+	desc = "A pistol-sized Colt assault carbine chambered for 5mm made by Olympic Arms, issued to Vertibird pilots. It's lighter, concealable and boasts a higher rate of fire - though the short barrel and lack of a buttstock make it a very poor choice for anything but boarding encounters."
+	icon_state = "olympic"
+	item_state = "assault_carbine"
+	icon_prefix = "olympic"
+	icon = 'icons/fallout/objects/guns/ballistic.dmi'
+	slowdown = 0.15
+	mag_type = /obj/item/ammo_box/magazine/m5mm
+	autofire_shot_delay = 0.9
+	spread = 30 //almost unusable
+	extra_damage = -2
+	can_attachments = FALSE
+	can_suppress = FALSE
+	can_flashlight = FALSE
+	recoil = 2
+	w_class = WEIGHT_CLASS_NORMAL
+	weapon_weight = WEAPON_HEAVY
+
+/obj/item/gun/ballistic/automatic/assault_carbine/smg
+	name = "\improper 5mm machine pistol"
+	desc = "A German-made armor piercing PDW chambered for 5mm. It has the markings of the Brotherhood of Steel on its receiver."
+	icon_state = "mp7"
+	item_state = "uzi"
+	icon_prefix = "mp7"
+	icon = 'icons/fallout/objects/guns/ballistic.dmi'
+	slowdown = 0.12
+	autofire_shot_delay = 1.2
```

Comment:
This is WAY too fast for 5mm weapons.

Reply:
I think it's fair given that it has a negative damage and AP bonus